modules-js/react-fleet/src/form-elements/inputs/TextInput.tsx

```diff
@@ -50,24 +64,56 @@ export default function TextInput(props: Props): JSX.Element {
     }`,
   };
 
+  const labelText = (label): React.ReactChild => {
+    if (typeof label === 'string') {
+      return (
+        <span style={{ marginRight: '0.5em', whiteSpace: 'nowrap' }}>
+          {label}
+        </span>
+      );
+    } else {
+      return label;
+    }
+  };
+
+  // If a React node is passed in for the input’s label, but “hideLabel” is
+  // true, the inner text is extracted from each child node so we have plain
+  // text for the “aria-label” attribute. Self-closing tags (e.g <wbr /> or
+  // <br />) are ignored. Child nodes may only contain plain text.
+  const extractAriaLabelText = (label): string => {
```

Comment:
It would be good to have a type on `label`

Reply:
TBH, though, is this necessary? You could instead use validation or a type union on `Props` to ensure that if `hideLabel` is true that `label` is just a `string`.

AFAICT it doesn’t make sense to provide a complex HTML label that’s explicitly hidden.

If you do keep this, I’d like to have a story that exercises it.